.github/workflows/update-ruff.yaml

```diff
@@ -0,0 +1,103 @@
+      with:
+        version: latest
+
+    - name: Update Ruff version in dev-requirements.txt
+      run: sed -i "s/ruff==.*/ruff==${{ steps.latest-version.outputs.ruff-version }}/" dev-requirements.txt
+
+    - name: Update Ruff minimum version in pyproject.toml
+      # Update required-version only in [tool.ruff] section
+      run: sed -i '/^\[tool\.ruff\]/,/^\[/ s/required-version = ">=.*"/required-version = ">=${{ steps.latest-version.outputs.ruff-version }}"/' pyproject.toml
+
+    - name: Run Ruff formatting with new version
+      run: make fmt-ruff
+
+    - name: Check if formatting passed
+      run: make lint-ruff
+
+    - name: Format PR title
+      id: format-title
+      run: |
+        if [ "${{ github.ref_name }}" != "development" ]; then
+          echo "title=[Requirements-automated] Update Ruff from ${{ steps.current-version.outputs.current }} to ${{ steps.latest-version.outputs.ruff-version }} [${{ github.ref_name }}]" >> "$GITHUB_OUTPUT"
+        else
+          echo "title=[Requirements-automated] Update Ruff from ${{ steps.current-version.outputs.current }} to ${{ steps.latest-version.outputs.ruff-version }}" >> "$GITHUB_OUTPUT"
+        fi
+
+    - uses: peter-evans/create-pull-request@271a8d0340265f705b14b6d32b9829c1cb33d45e # 7.0.8
+      with:
+        title: ${{ steps.format-title.outputs.title }}
+        body: |
+          🤖 **Automated Ruff Update**
+
+          This PR updates Ruff from version **${{ steps.current-version.outputs.current }}** to **${{ steps.latest-version.outputs.ruff-version }}**.
+
+          ## Changes
+          - Updated `dev-requirements.txt` with the latest Ruff version
+          - Updated `pyproject.toml` minimum Ruff version requirement
+          - Applied automatic formatting with `make fmt-ruff`
+          - Verified code passes linting with `make lint-ruff`
+
+          ## Testing
+          ✅ Code formatting completed successfully
+          ✅ Linting checks passed
+
+          Generated by the workflow run:
+          ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
+
+          ---
+          *This PR was created automatically by the monthly Ruff update workflow.*
+        commit-message: |
+          Update Ruff from ${{ steps.current-version.outputs.current }} to ${{ steps.latest-version.outputs.ruff-version }}
+
+          - Update dev-requirements.txt with Ruff ${{ steps.latest-version.outputs.ruff-version }}
+          - Update pyproject.toml minimum Ruff version requirement
+          - Apply automatic formatting with make fmt
+          - Verify linting passes with make lint
+        branch: "automated/ruff-update-${{ github.ref_name }}"
+        delete-branch: true
+        # See:
+        # https://github.com/peter-evans/create-pull-request?tab=readme-ov-file#token
+        # Fine-grained Personal Access Token (PAT) with contents: write and pull-requests: write scopes.
+        token: ${{ secrets.MLRUN_PR_PAT }}
```

Comment:
we no longer use mlrun_pr_pat but rather apps token. use @tal-haim to update your PR